qiskit/transpiler/preset_passmanagers/level0.py

```diff
@@ -77,16 +77,24 @@ def level_0_pass_manager(pass_manager_config: PassManagerConfig) -> StagedPassMa
     def _choose_layout_condition(property_set):
         return not property_set["layout"]
 
+    if coupling_map is None:
+        coupling_map_layout = target
+    else:
+        coupling_map_layout = coupling_map
```

Comment:
this seems to prevent target from actually being passed to the relevant passes. Because when I do

```py
transpile(circuit, target=target)
```
internally somewhere the coupling_map is built. And with this line here this means the passes are actually called with `coupling_map` and not with `target` (all the info in target get lost).

This is true for all levels 0-3.

Reply:
locally i fixed this by doing

```suggestion
    if target is None:
        coupling_map_layout = coupling_map
    else:
        coupling_map_layout = target
```

Reply:
Oh, that's fair we can default to the `Target` here like this. I did this because I was thinking of an issue we had recently where you couldn't override a backend's target via `transpile()` with the `coupling_map` argument. But that should be fixed in the `transpile()` level already so this is fine to do.